### DISCUSSION TITLE
N*N tiles

### DISCUSSION THREAD

**Discussion by FedeTress** [AlexeyBond/godot-constraint-solving#29] [Q&A]
Reading its documentation, i understood that mxgmn's original algorithm let you choose an array of N*N group of tiles to use as models for the program to create the map (so that it's locally similar to the pattern of the initial image itself); is there a way to use this kind of mode? At the moment it seems to me the only option is to constrain single tiles with what tiles can surround them, but if i compose an image it will not be locally similar to it. The main reason for this question is because i'm working with only two tiles, wall and floor, so the constraints are either none (i can place a wall near a floor andi vice versa) or everything (i cant, so my map is full of identical tiles)

**Comment by AlexeyBond** [comment]
Hi.

Check out this example: https://github.com/AlexeyBond/godot-constraint-solving/blob/master/addons/wfc/examples/demo_wfc_2d_tilemap_multitile.tscn
